src/client/content_cao.cpp

```diff
@@ -989,10 +989,13 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 			updateTextures(m_previous_texture_modifier);
 		}
 	}
-	if (!getParent() && std::fabs(m_prop.automatic_rotate) > 0.001) {
-		m_rotation.Y += dtime * m_prop.automatic_rotate * 180 / M_PI;
-		rot_translator.val_current = m_rotation;
-		updateNodePos();
+
+	if (getParent()) {
+		// This is the child node's rotation. It is only used for automatic_rotate.
+		v3f local_rot = node->getRotation();
+		local_rot.Y = modulo360f(local_rot.Y - dtime * core::RADTODEG *
+				(getParent() ? 0.f : m_prop.automatic_rotate));
```

Comment:
You already checked for `getParent()` with `if`.

Reply:
Discussed on IRC. This has been fixed in the latest push already. But thanks :)